fieldpath/set.go

```diff
@@ -136,6 +138,126 @@ func (s *Set) EnsureNamedFieldsAreMembers(sc *schema.Schema, tr schema.TypeRef)
 	}
 }
 
+// MustPrefixPattern is the same as PrefixPattern except it panics if parts can't be
+// turned into a SetPattern.
+func MustPrefixPattern(parts ...interface{}) *SetPattern {
```

Comment:
I'd argue for "Make" in the names of these.

Reply:
`MakePrefixPatternOrDie` similar to `MakePathOrDie` in other file? 

https://github.com/kubernetes-sigs/structured-merge-diff/blob/f034ec0f9a297cb014e5b5f7f4b4c9e4545777eb/fieldpath/path.go#L112

Reply:
I've gone with `MakePrefixMatcherOrDie` to conform to the conventions of the most adjacent code.  Feel free to argue for a different convention.